app/helpers/application_helper.rb

```diff
@@ -58,4 +58,19 @@ def mobile_device?
     end
     false
   end
+
+  def add_user_to_anthology(user, anth)
+    unless user.rep_group_list.include?(anth.slug)
+      user.rep_group_list.add(anth.slug)
+
+      # fix validation error by checking agreement box
+      user.agreement = true
```

Comment:
This seems like it belongs in the user creation step, not the add_user_to_anthology step, doesn't it?

Reply:
I think the original idea was the user would agree to the terms and conditions when they make an account. As a result, the `User` model fails validation when trying to save it if `agreement = false`.

But as other account-related features were added, such as SAML login, there are cases where a user record is created without showing the user that form. For these cases, the `agreement` field is automatically set to `true` without asking the user. See the `find_for_external_authentication` method in the User model as an example.

I could change the default value to `true`, bypassing the need for special handling, but at that point I don't know why we'd ask about terms and conditions at all. It's probably out of scope for this bug fix PR, but simply removing the `agreement` field from the User model would make the most sense at this point - the vast majority of user accounts today are created with external authentication which never shows them the COVE T&C anyway.